osf/models/registrations.py

```diff
@@ -189,6 +189,18 @@ def archiving(self):
         job = self.archive_job
         return job and not job.done and not job.archive_tree_finished()
 
+    @property
+    def date_withdrawn(self):
+        if self.root.retraction is None:
+            return None
+        return self.root.retraction.date_retracted
```

Comment:
My opinion, this and `withdrawal_justification` below, could be made more readable. Something like:
```python
@property
def date_withdrawn(self):
   return getattr(self.root.retraction, 'date_retracted', None)
````